[SDK 50] Invalid character escape when using npx expo run android on windows

### Minimal reproducible example

https://github.com/smashboy/expo-sdk-50-escape-char-bug

### Summary

Hello everyone! Im trying to upgrade to sdk 50, so I can use latest expo-sqlite and currently build fails with the following error after running `npx expo run android`

Steps to reproduce:

1. C:\projects> npx create-expo-app -t blank@beta my-app 
2. C:\projects> cd my-app 
3. C:\projects\my-app> npx expo install expo-sqlite@next 
4. C:\projects\my-app> npx expo run:android

```
Task :expo-sqlite:configureCMakeDebug[arm64-v8a]
C/C++: CMake Error at CMakeLists.txt:13 (add_library):
C/C++:   Syntax error in cmake code when parsing string
C/C++:     C:\projects\flowappnative\node_modules\expo-sqlite\android\build\sqlite3_src/sqlite3.c
C/C++:   Invalid character escape '\p'.

> Task :expo-sqlite:configureCMakeDebug[arm64-v8a] FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':expo-sqlite:configureCMakeDebug[arm64-v8a]'.
> [CXX1429] error when building with cmake using C:\projects\flowappnative\node_modules\expo-sqlite\android\CMakeLists.txt: -- The C compiler identification is Clang 14.0.6
  -- The CXX compiler identification is Clang 14.0.6
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: C:/Users/User/AppData/Local/Android/Sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/bin/clang.exe - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: C:/Users/User/AppData/Local/Android/Sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/bin/clang++.exe - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Configuring incomplete, errors occurred!
  See also "C:/projects/flowappnative/node_modules/expo-sqlite/android/.cxx/Debug/16186y06/arm64-v8a/CMakeFiles/CMakeOutput.log".

  C++ build system [configure] failed while executing:
      @echo off
      "C:\\Users\\User\\AppData\\Local\\Android\\Sdk\\cmake\\3.22.1\\bin\\cmake.exe" ^
        "-HC:\\projects\\flowappnative\\node_modules\\expo-sqlite\\android" ^
        "-DCMAKE_SYSTEM_NAME=Android" ^
        "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON" ^
        "-DCMAKE_SYSTEM_VERSION=21" ^
        "-DANDROID_PLATFORM=android-21" ^
        "-DANDROID_ABI=arm64-v8a" ^
        "-DCMAKE_ANDROID_ARCH_ABI=arm64-v8a" ^
        "-DANDROID_NDK=C:\\Users\\User\\AppData\\Local\\Android\\Sdk\\ndk\\25.1.8937393" ^
        "-DCMAKE_ANDROID_NDK=C:\\Users\\User\\AppData\\Local\\Android\\Sdk\\ndk\\25.1.8937393" ^
        "-DCMAKE_TOOLCHAIN_FILE=C:\\Users\\User\\AppData\\Local\\Android\\Sdk\\ndk\\25.1.8937393\\build\\cmake\\android.toolchain.cmake" ^
        "-DCMAKE_MAKE_PROGRAM=C:\\Users\\User\\AppData\\Local\\Android\\Sdk\\cmake\\3.22.1\\bin\\ninja.exe" ^
        "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=C:\\projects\\flowappnative\\node_modules\\expo-sqlite\\android\\build\\intermediates\\cxx\\Debug\\16186y06\\obj\\arm64-v8a" ^
        "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY=C:\\projects\\flowappnative\\node_modules\\expo-sqlite\\android\\build\\intermediates\\cxx\\Debug\\16186y06\\obj\\arm64-v8a" ^
        "-DCMAKE_BUILD_TYPE=Debug" ^
        "-DCMAKE_FIND_ROOT_PATH=C:\\projects\\flowappnative\\node_modules\\expo-sqlite\\android\\.cxx\\Debug\\16186y06\\prefab\\arm64-v8a\\prefab" ^ 
        "-BC:\\projects\\flowappnative\\node_modules\\expo-sqlite\\android\\.cxx\\Debug\\16186y06\\arm64-v8a" ^
        -GNinja ^
        "-DANDROID_STL=c++_shared" ^
        "-DSQLITE3_SRC_DIR=C:\\projects\\flowappnative\\node_modules\\expo-sqlite\\android\\build\\sqlite3_src"
    from C:\projects\flowappnative\node_modules\expo-sqlite\android
  CMake Error at CMakeLists.txt:13 (add_library):
    Syntax error in cmake code when parsing string

      C:\projects\flowappnative\node_modules\expo-sqlite\android\build\sqlite3_src/sqlite3.c

    Invalid character escape '\p'.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 5m 8s
489 actionable tasks: 143 executed, 316 from cache, 30 up-to-date
Error: C:\projects\flowappnative\android\gradlew.bat exited with non-zero code: 1
Error: C:\projects\flowappnative\android\gradlew.bat exited with non-zero code: 1
    at ChildProcess.completionListener (C:\projects\flowappnative\node_modules\@expo\cli\node_modules\@expo\spawn-async\build\spawnAsync.js:52:23)
    at Object.onceWrapper (node:events:628:26)
    at ChildProcess.emit (node:events:513:28)
    at cp.emit (C:\projects\flowappnative\node_modules\@expo\cli\node_modules\cross-spawn\lib\enoent.js:34:29)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5)
    ...
    at Object.spawnAsync [as default] (C:\projects\flowappnative\node_modules\@expo\cli\node_modules\@expo\spawn-async\build\spawnAsync.js:17:21)    
    at spawnGradleAsync (C:\projects\flowappnative\node_modules\@expo\cli\build\src\start\platforms\android\gradle.js:72:46)
    at Object.assembleAsync (C:\projects\flowappnative\node_modules\@expo\cli\build\src\start\platforms\android\gradle.js:52:18)
    at runAndroidAsync (C:\projects\flowappnative\node_modules\@expo\cli\build\src\run\android\runAndroidAsync.js:36:24)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Environment

 expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 18.6.0 - C:\Program Files\nodejs\node.EXE        
      Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 8.13.2 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: AI-231.9392.1.2311.11076708
    npmPackages:
      expo: ^50.0.0-preview.7 => 50.0.0-preview.7
      react: 18.2.0 => 18.2.0
      react-native: 0.73.1 => 0.73.1
    Expo Workflow: bare

I've got an similar error.

**> Task :expo-sqlite:configureCMakeDebug[arm64-v8a] FAILED**
C/C++: CMake Error at CMakeLists.txt:13 (add_library):
C/C++:   Syntax error in cmake code when parsing string
C/C++:     C:\apps\kitab\node_modules\expo-sqlite\android\build\sqlite3_src/sqlite3.c
C/C++:   Invalid character escape '\w'.

--- a/packages/expo-sqlite/CHANGELOG.md
+++ b/packages/expo-sqlite/CHANGELOG.md
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Fixed building error on Windows. ([#26296](https://github.com/expo/expo/pull/26296) by [@kudo](https://github.com/kudo))
+
 ### 💡 Others
 
 ## 13.1.2 - 2023-12-21

--- a/packages/expo-sqlite/android/build.gradle
+++ b/packages/expo-sqlite/android/build.gradle
@@ -1,3 +1,5 @@
+import org.apache.tools.ant.taskdefs.condition.Os
+
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 apply plugin: 'maven-publish'
@@ -15,6 +17,14 @@ if (expoModulesCorePlugin.exists()) {
     useExpoPublishing()
     useCoreDependencies()
   }
+}
+
+String toPlatformIndependentPath(File path) {
+  def result = path.toString()
+  if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+    result = result.replace(File.separatorChar, '/' as char)
+  }
+  return result
 }
 
 def SQLITE_VERSION = '3420000'
@@ -112,7 +122,7 @@ android {
       cmake {
         abiFilters (*reactNativeArchitectures())
         arguments "-DANDROID_STL=c++_shared",
-          "-DSQLITE3_SRC_DIR=${SQLITE3_SRC_DIR}"
+          "-DSQLITE3_SRC_DIR=${toPlatformIndependentPath(SQLITE3_SRC_DIR)}"
       }
     }
   }

